package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -55,7 +79,95 @@ def to_json(self) -> str:
         with self._db_session_class.begin() as session:
-            session.add(RunModel(id=self._session_id, blob=self.to_json()))
+            session.add(RunModel(id=self._session_id, blob=self._to_json()))
+        if self.remote_location:
+            self._upload()
+
+    def _upload(self):
+        """Uploads the session store database file to the specified
+        remote path on the cloud storage."""
+        db_name = _get_dbname()
+        logger.debug(f"Uploading local session store to remote with name {db_name}...")
+        try:
+            self._remote_fs.put(self.location, f"{self.remote_location}/{db_name}")
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _download(self):
+        """Downloads all the session store database files
+        from the specified remote path on the cloud storage
+        to your local project.
+        """
+        try:
+            # In theory we should be able to do this as a single operation:
+            # self._remote_fs.get(f"{self.remote_location}/*.db", str(Path(self.location).parent))
+            # but this does not seem to work correctly - maybe a bug in fsspec. So instead
+            # we do it in two steps.
+            remote_dbs = self._remote_fs.glob(f"{self.remote_location}/*.db")
+            logger.debug(f"Downloading {len(remote_dbs)} remote session stores to local...")
+            self._remote_fs.get(remote_dbs, str(Path(self.location).parent))
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _merge(self):
+        """Merges all the session store databases stored at the
+        specified locations into the user's local session_store.db
+
+        Notes:
+        - This method uses multiple SQLAlchemy engines to connect to the
+        user's session_store.db and to all the other downloaded dbs.
+        - It is assumed that all the databases share the same schema.
+        - In the version 1.0 - we only merge the runs table which
+        contains all the experiments.
+        """
+
+        all_new_runs = []
+
+        with self._db_session_class() as session:
+            existing_run_ids = session.execute(select(RunModel.id)).scalars().all()
+
+        # Look at all databases in the local session store directory
+        # that aren't the actual session_store.db itself.
+        downloaded_db_locations = set(Path(self.location).parent.glob("*.db")) - {
+            Path(self.location)
+        }
+
+        logger.debug(f"Checking {len(downloaded_db_locations)} downloaded session stores for new runs...")
+        for downloaded_db_location in downloaded_db_locations:
+            engine = create_engine(f"sqlite:///{downloaded_db_location}")
+            with Session(engine) as session:
+                new_runs = (
+                    session.query(RunModel)
+                    .filter(RunModel.id.not_in(existing_run_ids))
+                    .all()
```

Comment:
I don't like using `session.query` here (compared to `session.execute` above), but I know it matches what we have in runs.py so it's fine for now. We should update it as part of https://github.com/kedro-org/kedro-viz/issues/1360.